File: python/selenium/python3.6/iOSAppTest.py
from appium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.remote.remote_connection import RemoteConnection
import unittest
import time
import sys
sys.path.append('../..')
from python import configs

class iOSAppTest(unittest.TestCase):

  def setUp(self):
    self._command_executor = RemoteConnection(configs.kobitonServerUrl, resolve_ip=False)
    self._command_executor.set_timeout(configs.session_timeout)
    self.driver = webdriver.Remote(configs.kobitonServerUrl, configs.desired_caps_ios_app)

  def tearDown(self):
    self.driver.quit()

  def test_ios_app(self):
    self.search_questions_on_Acura_Support_Community()
    self.search_IFixit_on_home_screen()

  def search_questions_on_Acura_Support_Community(self):
    print ('should allow to navigate to some devices on Acura Support Community')
    '''
    Steps:
    1. Click on "Car and Truck" Categories on Homepage
    2. Click on "Acura" Categories

    Expected:
    1. General Information is "Acura".
    2.Verify five devices below displays.
    + Acura Integra
    + Acura MDX
    + Acura RL
    + Acura TL
    + Acura TSX
    '''
    self.driver.find_element_by_xpath("//XCUIElementTypeButton[@name='START A REPAIR']").click()
    time.sleep(5)
    self.driver.find_element_by_xpath("//*[@name='Car and Truck']").click()
    time.sleep(2)
    self.driver.find_element_by_xpath("//*[@name='Acura']").click()
    time.sleep(2)
    WebDriverWait(self.driver, 60).until(
    EC.element_to_be_clickable((By.XPATH, "//XCUIElementTypeNavigationBar")))
    acura_text = self.driver.find_element_by_xpath("//XCUIElementTypeNavigationBar").get_attribute('name')

    has_acura_integra = self.driver.find_element_by_xpath("//XCUIElementTypeStaticText[@name='Acura Integra']").is_displayed()
    has_acura_MDX = self.driver.find_element_by_xpath("//XCUIElementTypeStaticText[@name='Acura MDX']").is_displayed()
    has_acura_RL = self.driver.find_element_by_xpath("//XCUIElementTypeStaticText[@name='Acura RL']").is_displayed()
    has_acura_TL = self.driver.find_element_by_xpath("//XCUIElementTypeStaticText[@name='Acura TL']").is_displayed()
    has_acura_TSX = self.driver.find_element_by_xpath("//XCUIElementTypeStaticText[@name='Acura TSX']").is_displayed()

    self.assertEqual(acura_text, 'Acura')
    self.assertTrue(has_acura_integra)
    self.assertTrue(has_acura_MDX)
    self.assertTrue(has_acura_RL)
    self.assertTrue(has_acura_TL)
    self.assertTrue(has_acura_TSX)

  def search_IFixit_on_home_screen(self):
    print ('should allow to search iFixit on Home screen')
    self.driver.launch_app()
    time.sleep(2)
    self.driver.find_element_by_xpath("//XCUIElementTypeButton[@name='START A REPAIR']").click()
    time.sleep(2)
    self.driver.find_element_by_xpath("//*[@name='Search']").click()
    self.driver.find_element_by_xpath("//XCUIElementTypeSearchField[@name='Search']").send_keys('Macbook Pro 2015')

    time.sleep(2)

    firstListResult = self.driver.find_elements_by_xpath("//XCUIElementTypeStaticText[contains(@label,'MacBook Pro')]")

    self.driver.find_element_by_xpath("//XCUIElementTypeButton[@name='Cancel']").click()
    self.driver.find_element_by_xpath("//*[@name='Search']").click()
    self.driver.find_element_by_xpath("//XCUIElementTypeSearchField[@name='Search']").send_keys('Acura')
    self.driver.find_element_by_xpath("//XCUIElementTypeButton[@name='Categories']").click()
    time.sleep(2)

    secondListResult = self.driver.find_elements_by_xpath("//XCUIElementTypeStaticText[contains(@label,'Acura')]")

    self.assertTrue(len(firstListResult) >= 33, 'The expected results are greater or equal to 33 results.')
    self.assertTrue(len(secondListResult) >= 6, 'The expected results are greater or equal to 6 results.')

if __name__ == '__main__':
    suite = unittest.TestLoader().loadTestsFromTestCase(iOSAppTest)
    unittest.TextTestRunner(verbosity=2).run(suite)

File: ruby/test/iOSAppTest.rb
require 'rubygems'
require 'test/unit'
require 'appium_lib'
require 'selenium-webdriver'
require './configs.rb'
extend Test::Unit::Assertions

class IOSAppTest < Test::Unit::TestCase

  def setup
    desire_cap = DesiredCapabilities.new
    @driver = Appium::Driver.new(desire_cap.desired_caps_ios_app)
    @driver.start_driver.manage.timeouts.implicit_wait = 120
    @wait = Selenium::WebDriver::Wait.new(:timeout => 120)
  end

  def teardown
    if !@driver.nil?
      begin
        @driver.driver_quit
      rescue Exception => e
        puts e.message
      ensure
        @driver.driver_quit
      end
    end
  end
  
  def test_iFixit_iOS_app
    verify_navigation_on_acura_support_community
    verify_search_iFixit_on_home_screen
  end

  def verify_navigation_on_acura_support_community
    puts 'Test case 1: it should allow to navigate to some devices on Acura Support Community'

=begin
     Steps:
      1. Click on "Car and Truck" Categories on Homepage
      2. Click on "Acura" Categories
     Expected:
      1. General Information is "Acura".
      2.Verify five devices below displays.
      + Acura Integra
      + Acura MDX
      + Acura RL
      + Acura TL
      + Acura TSX
=end

    @driver.find_element(:xpath, "//XCUIElementTypeButton[@name='START A REPAIR']").click
    sleep 2
    @driver.find_element(:xpath, "//*[@name='Car and Truck']").click
    @driver.find_element(:xpath, "//*[@name='Acura']").click
    sleep 2
    @wait.until {@driver.find_element(:xpath, "//XCUIElementTypeNavigationBar").displayed?}
    acura_text = @driver.find_element(:xpath, "//XCUIElementTypeNavigationBar").attribute("name")
    has_acura_integra = @driver.find_element(:xpath, "//XCUIElementTypeStaticText[@name='Acura Integra']").displayed?
    has_acura_mdx = @driver.find_element(:xpath, "//XCUIElementTypeStaticText[@name='Acura MDX']").displayed?
    has_acura_rl = @driver.find_element(:xpath, "//XCUIElementTypeStaticText[@name='Acura RL']").displayed?
    has_acura_tl = @driver.find_element(:xpath, "//XCUIElementTypeStaticText[@name='Acura TL']").displayed?
    has_acura_tsx = @driver.find_element(:xpath, "//XCUIElementTypeStaticText[@name='Acura TSX']").displayed?
    @driver.close_app
    assert_equal(acura_text, 'Acura')
    assert_equal(has_acura_integra, true)
    assert_equal(has_acura_mdx, true)
    assert_equal(has_acura_rl, true)
    assert_equal(has_acura_tl, true)
    assert_equal(has_acura_tsx, true)
  end

  def verify_search_iFixit_on_home_screen
    puts 'Test case 2: it should allow to search iFixit on Home screen'
    @driver.launch_app
    sleep 2
    @driver.find_element(:xpath, "//XCUIElementTypeButton[@name='START A REPAIR']").click
    sleep 2
    @wait.until {@driver.find_element(:xpath, "//XCUIElementTypeSearchField[@name='Search']").displayed?}
    @driver.find_element(:xpath, "//XCUIElementTypeSearchField[@name='Search']").send_keys('Macbook Pro 2015')
    sleep 2
    first_result = @driver
        .find_elements(:xpath, "//XCUIElementTypeStaticText[contains(@label,'MacBook Pro')]")
    @driver.find_element(:xpath, "//XCUIElementTypeButton[@name='Cancel']").click
    @driver.find_element(:xpath, "//*[@name='Search']").click
    @driver.find_element(:xpath, "//XCUIElementTypeSearchField[@name='Search']").send_keys('Acura')
    @driver.find_element(:xpath, "//XCUIElementTypeButton[@name='Categories']").click
    sleep 2
    second_result = @driver
        .find_elements(:xpath, "//XCUIElementTypeStaticText[contains(@label,'Acura')]")
    assert_true(first_result.size >= 33, "The expected results are greater or equal to 33 results but the actual results are #{first_result}.")
    assert_true(second_result.size >= 6, "The expected results are greater or equal to 6 results but the actual results are #{second_result}.")
  end
  
end
